File: 7.Go版每日一题/789. 逃脱阻碍者.ts
function escapeGhosts(ghosts: number[][], target: number[]): boolean {
	const start = [0, 0]
	const minD = getDistance(start, target)
	for (const g of ghosts) {
		if (getDistance(g, target) <= minD) {
			return false
		}
	}
	return true
}

const getDistance = (point1: number[], point2: number[]): number => Math.abs(point1[0] - point2[0]) + Math.abs(point1[1] - point2[1])

File: 7.Go版每日一题/789. 逃脱阻碍者.go
package main

func main() {}

func escapeGhosts(ghosts [][]int, target []int) bool {
	myPoint := []int{0, 0}
	d := getDistance(myPoint, target)
	for _, ghost := range ghosts {
		if getDistance(ghost, target) <= d {
			return false
		}
	}
	return true
}

func getDistance(myPoint, point2 []int) int {
	return abs(myPoint[0]-point2[0]) + abs(myPoint[1]-point2[1])
}

func abs(x int) int {
	if x < 0 {
		return -x
	}
	return x
}
